src/rules/label-has-for.js

```diff
@@ -8,7 +8,7 @@ const {
 
 const controlTypes = ["input", "meter", "progress", "select", "textarea"];
 
-const validateNesting = (node, allowChildren) =>
+const validateNesting = (node, allowChildren, controlComponents) =>
```

Comment:
What do you think about having the controlComponents and the allowChildren both on the same second parameter as an object?

```
const validateNesting = (node, options) => {
  const { allowChildren, controlComponents } = options;
```

This way it would be easier to do the recursion and it would be more convinient to pass the arguments on the call stack on the rest of the file, meaning we would only need to write both, the allowChildren, and the controlComponents on the first call in the stack, the `isValidLabel(node, required, { allowChildren, controlComponents })`, the rest would only reference this options by "options"